cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -225,63 +168,31 @@ function deploy_sriov_operator {
 
   echo 'Generating webhook certificates for the SR-IOV operator webhooks'
   pushd "${CERTCREATOR_PATH}"
-    go run . -namespace sriov-network-operator -secret operator-webhook-service -hook operator-webhook -kubeconfig $KUBECONFIG_PATH || return 1
```

Comment:
please inject here a false, i think that it won't return (according what i tried, unless i am wrong)

Reply:
config_sriov.sh runs with `-e` flag, if `certcreator` will raise an error this script will be stopped,
this is already been tested and the script actually stops.

using `|| return 1` after operations inside helpers and `|| exit 1` on calling for helpers, interfere with bash catching the errors although `-e` flag is up, this is why it used to continue on errors.

Reply:
Please see here
https://github.com/kubevirt/kubevirtci/pull/481/files#diff-499cff045d8a875e8b39f4a9fd59354eb99e8f8d91a95ebc70fbf42d2d5b5beaR275

This one is a function, in the same file, and when there was the bug that that PR solved (locally is saw it),
it continued even that the command failed, so i would suggest to inject error in all the changed places and understand exactly what happens.

Reply:
I tested it on a local environment and saw that the script stops if there is an error inside or by the helpers functions

Reply:
imo please test each of them please.
as i also saw problems locally for sure, at least on some of them.

Reply:
As I said it already been tested, if you have other insights that suggest a change please share it.

Reply:
I just re-simulated it locally (see diff in the PR please), created this PR to see if it happens on CI
https://github.com/kubevirt/kubevirtci/pull/494

Reply:
please see that it continue even that the command failed, also on CI
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirtci/494/check-up-kind-1.17-sriov/1334144011677470720#1:build-log.txt%3A1860

Reply:
see https://github.com/kubevirt/kubevirtci/pull/492#discussion_r537452325

Reply:
Since you said you checked and it stops if there is an error in a function,
but i saw the two behaviors, i raised it again with an example.

Lets be synced on the action items please.
We can't just remove `return 1` and the `exit 1`,
Agree ?

Reply:
Gotcha, please take another look on the thread that I refer to I answered about the `return 1` and `exit 1`,

Reply:
I took and answered.
My comment stands
here is a proof on the non retry functions:
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirtci/494/check-up-kind-1.17-sriov/1334144011677470720#1:build-log.txt%3A1860

Reply:
It dosent proof what you say, try it again with this PR changes and you will see it stops.
Since we are using `-e` flag there is no need to add `|| return/exit 1` (unless you want to consider only a return code of a function) it was a mistake to add it in the first place.

Reply:
This is after you fixed, before that it was broken.

Reply:
I removed `|| return/exit 1` from the beginning so it would work for you if you would check with the PR changes.

Reply:
This was the fix that fixed it from what i know, and it was after the example of this thread.
https://github.com/kubevirt/kubevirtci/compare/7c4c152fe237945e0654b55e8b350d617ea32201..3229c3e908c267e13576224de7583645dec652da

The removal of the return/exit didnt fix it, the changing of the set fixed it.

Reply:
The commit that adds explicit `set -e` actually has no affect, since it is already configured by sourcing `kind/common.sh` at the beginning of the script (it is configured there).
Setting that flag with `#!/bin/bash -e` will not take affect.

I added this commit so it will be explicit that we configure it on that script instead of relaying on other files.
I also mentioned it in the commit message.